rotkehlchen/chain/evm/decoding/hop/decoder.py

```diff
@@ -0,0 +1,141 @@
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.chain.ethereum.utils import (
+    token_normalized_value,
+    token_normalized_value_decimals,
+)
+from rotkehlchen.chain.evm.decoding.hop.constants import CPT_HOP, HOP_CPT_DETAILS, HopBridge
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.constants.assets import A_ETH
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress, EvmTransaction
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.chain.evm.structures import EvmTxReceiptLog
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+TRANSFER_FROM_L1_COMPLETED = b'2\tX\x17i0\x80N\xb6l#C\xc74?\xc06}\xc1bIY\x0c\x0f\x19W\x83\xbe\xe1\x99\xd0\x94'  # noqa: E501
+WITHDRAWAL_BONDED = b'\x0c=%\x0cx1\x05\x1ex\xaajVg\x9eY\x03t\xc7\xc4$A_\xfeJ\xa4tI\x1d\xef/\xe7\x05'  # noqa: E501
+
+
+class HopCommonDecoder(DecoderInterface):
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            bridges: dict[ChecksumEvmAddress, HopBridge],
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.bridges = bridges
+        self.bridge_contracts = {bridge.address for bridge in self.bridges.values()}
+
+    def _decode_bridge(
+            self,
+            transaction: 'EvmTransaction',
+            decoded_events: list['EvmEvent'],
+            all_logs: list['EvmTxReceiptLog'],
+    ) -> list['EvmEvent']:
+        """This function is used to decode the brdiging events done via Hop. It checks if already
+        decoded events interacted with the Hop Protocol, and overwrites them if they did."""
+        for _log in all_logs:
+            if _log.topics[0] in WITHDRAWAL_BONDED:
+                self._decode_withdrawal_bonded(context=DecoderContext(
+                    tx_log=_log,
+                    transaction=transaction,
+                    action_items=[],
+                    all_logs=all_logs,
+                    decoded_events=decoded_events,
+                ))
+        return decoded_events
```

Comment:
why the underscore in from of the variable?

```suggestion
        for log in all_logs:
            if log.topics[0] in WITHDRAWAL_BONDED:
                self._decode_withdrawal_bonded(context=DecoderContext(
                    tx_log=log,
                    transaction=transaction,
                    action_items=[],
                    all_logs=all_logs,
                    decoded_events=decoded_events,
                ))
        return decoded_events
```

Reply:
because of the `log = RotkehlchenLogsAdapter(logger)` in line 33